### PR TITLE
Job artifact upload transform for Girder

### PR DESCRIPTION
**Pull request by zachmullen**

### CODE REVIEW

**Comment by zachmullen** [comment]
@kotfic PTAL, if you like this in concept I can add any necessary tests.

File: girder_worker_utils/transforms/girder_io.py [hunk 1]
```diff
@@ -117,3 +119,34 @@ def cleanup(self):
...
+            self.gc.post('job/%s/artifact?%s' % (self.job_id, qs), data=fh)
+
+    def transform(self, path):
+        if self.job_id is None:
+            self.job_id = str(self.job['_id'])
+
+        if os.path.isdir(path):
+            for f in os.listdir(path):
+                f = os.path.join(path, f)
+                if os.path.isfile(f):
```
Comment (kotfic) [inline girder_worker_utils/transforms/girder_io.py]: Won't this flat-out ignore nested directories? That's fine if that is the intended functionality,  but either a comment or something in the class docstring would be great making that explicit.

**Comment by kotfic** [comment]
Other than adding a little bit of documentation wrt. uploading directories of artifacts this looks good to me

File: girder_worker_utils/transforms/girder_io.py (hunk 1 above)
Comment (zachmullen) [reply girder_worker_utils/transforms/girder_io.py]: It is intentional, for now. I can add a comment to make that clear.

**Comment by kotfic** [comment]
yeah if you wanted to add a test or two i'm happy to approve this

**Comment by zachmullen** [comment]
@kotfic this is ready for final review.

**Pull request merged**